Updated installation, updation, migration docs (#244)
rancher/longhorn-product-docs: docs/version-1.9/modules/en/pages/installation-setup/installation/airgapped-environment.adoc

:current-version: {page-component-version}
:doctype: book

{longhorn-product-name} can be installed in an air gapped environment by using a Helm chart, or the Rancher UI.

== Prerequisites

* https://helm.sh/docs/[Helm] v3.0 or later is required.
* Deploy {longhorn-product-name} components images to your own registry.
* Deploy Kubernetes CSI driver components images to your own registry.

== Image Management for Air-Gapped Environments

. **Obtain the list of required images** +
Run the script to download the complete list of all images required for {longhorn-product-name}. The script will generate a `longhorn-images.txt` file containing the list of images.
+
[,bash]
----

> #!/usr/bin/env bash
> CHART_REF=oci://dp.apps.rancher.io/charts/suse-storage:{current-version}
> set -euo pipefail
> setopt shwordsplit 2>/dev/null || true
> # Check for dependencies
> for cmd in yq jq helm tar; do
>   if ! command -v "$cmd" >/dev/null 2>&1; then
>     echo >&2 "Error: '$cmd' is required but not installed."
>     exit 1
>   fi
> done
> tmp_dir=$(mktemp -d)
> cleanup() {
>   rm -rf "$tmp_dir"
> }
> trap cleanup EXIT
> helm pull "$CHART_REF" --untar --untardir "$tmp_dir"
> # Pull and extract chart
> helm pull "$CHART_REF" --untar --untardir "$tmp_dir"
> # The folder name is just the chart name, not the version
> chart_dir="$tmp_dir/suse-storage"
> # Extract 'helm.sh/images' annotation and convert to JSON, then print image list (one per line)
> yq eval '.annotations."helm.sh/images"' "$chart_dir/Chart.yaml" | yq -o=json eval - | jq -r '.[] | .image' | sort > longhorn-images.txt

Reviewer: ```suggestion
#!/usr/bin/env bash
CHART_REF=oci://dp.apps.rancher.io/charts/suse-storage:{current-version}

set -euo pipefail
setopt shwordsplit 2>/dev/null || true

# Check for dependencies
for cmd in yq jq helm tar; do
  if ! command -v "$cmd" >/dev/null 2>&1; then
    echo >&2 "Error: '$cmd' is required but not installed."
    exit 1
  fi
done
tmp_dir=$(mktemp -d)
cleanup() {
  rm -rf "$tmp_dir"
}
trap cleanup EXIT

# Pull and extract chart
helm pull "$CHART_REF" --untar --untardir "$tmp_dir"

# The folder name is just the chart name, not the version
chart_dir="$tmp_dir/suse-storage"

# Extract 'helm.sh/images' annotation and convert to JSON, then print image list (one per line)
yq eval '.annotations."helm.sh/images"' "$chart_dir/Chart.yaml" | yq -o=json eval - | jq -r '.[] | .image' | sort > longhorn-images.txt
```

. **Load and push {longhorn-product-name} images to your private registry** + 
{longhorn-product-name} provides another script, https://raw.githubusercontent.com/longhorn/longhorn/v{current-version}/scripts/load-images.sh[`load-images.sh` script], to push images to your private container registry. To load images from a `tar.gz` file (for example, `longhorn-images.tar.gz`) and push them to your registry, run the following commands. Replace `<YOUR-PRIVATE-REGISTRY>` with the actual address of your private registry.
+
[,bash]
----
wget https://raw.githubusercontent.com/longhorn/longhorn/v{current-version}/scripts/load-images.sh
chmod +x load-images.sh
./load-images.sh --image-list longhorn-images.txt --images longhorn-images.tar.gz --registry <YOUR-PRIVATE-REGISTRY>
----
+
If you do not specify a `tar.gz` file using the `--images` flag, the script will find images in your local Docker cache and push them to the registry.

. **View script options**: + 
For more options when using these scripts, see their respective `--help` flags:
+
[,bash]
----
./save-images.sh --help
./load-images.sh --help
----

> == Using a Helm Chart

Reviewer: Rename to `== Installing by Helm Chart`

:current-version: {page-component-version}
:doctype: book

{longhorn-product-name} can be installed in an air-gapped environment by using a Helm chart.

== Prerequisites

* https://helm.sh/docs/[Helm] v3.0 or later is required.
* Deploy {longhorn-product-name} components images to your own registry.
* Deploy Kubernetes CSI driver components images to your own registry.

== Image Management for Air-Gapped Environments

. **Obtain the list of required images** +

> Run the link:{attachmentsdir}/v1.9.1-longhorn-image-list.sh[script] to download the complete list of all images required for {longhorn-product-name}. The script will generate a `longhorn-images.txt` file containing the list of images.

Reviewer 2: Rename `v1.9.1-longhorn-image-list.sh` to `download-longhorn-image-list.sh`

[,yaml]
----
privateRegistry:
  # -- Setting that allows you to create a private registry secret.
  createSecret: true
  # -- URL of a private registry. When unspecified, Longhorn uses the default system registry.
  registryUrl: <REGISTRY_URL>
  # -- User account used for authenticating with a private registry.
  registryUser: <REGISTRY_USER>
  # -- Password for authenticating with a private registry.
  registryPasswd: <REGISTRY_PASSWORD>
  # -- Kubernetes secret that allows you to pull images from a private registry. This setting applies only when creation of private registry secrets is enabled. You must include the private registry name in the secret name.
  registrySecret: <REGISTRY_SECRET_NAME>
----

| Using Custom Image Names
|
In `values.yaml`, configure the image settings for each component.

[NOTE]
====
Do not include the private registry prefix (for example, `example.com/username/`); it will be added automatically. If your image is `example.com/username/longhorn-manager`, use `username/longhorn-manager` in the following charts.
====

Specify the `repository` and `tag` for each {longhorn-product-name} component image:

[,yaml]
----
  image:
    longhorn:
      engine:

>         repository: containers/longhorn-engine

Reviewer 3: > Why does this differ from upstream

We source our images from the App Collection registry rather than from Longhorn’s upstream registry. In air‑gap installations we mirror those images into the air‑gap registry under the same name, hence the repository path differs.

:current-version: {page-component-version}
:doctype: book

{longhorn-product-name} can be installed in an air-gapped environment by using a Helm chart.

== Prerequisites

* https://helm.sh/docs/[Helm] v3.0 or later is required.
* Deploy {longhorn-product-name} components images to your own registry.
* Deploy Kubernetes CSI driver components images to your own registry.

== Image Management for Air-Gapped Environments

. **Obtain the list of required images** +
Run the link:{attachmentsdir}/download-longhorn-image-list.sh[script] to download the complete list of all images required for {longhorn-product-name}. The script will generate a `longhorn-images.txt` file containing the list of images.

[NOTE]
====

> Authentication to AppCo is necessary when running this script, as it uses Helm internally. For more information, refer to the https://docs.apps.rancher.io/get-started/authentication/#helm[Helm authentication documentation].

Reviewer: ```suggestion
This script pulls the SUSE Storage chart from the SUSE Application Collection OCI registry to extract the required image list. Helm must be configured with authentication credentials to access the registry. For setup instructions, refer to the [Helm authentication documentation](https://docs.apps.rancher.io/get-started/authentication/#helm).
```

Reviewer: Briefly introduce the reason for the required authentication, then add the authentication doc.

:current-version: {page-component-version}
:doctype: book

{longhorn-product-name} can be installed in an air-gapped environment by using a Helm chart.

== Prerequisites

* https://helm.sh/docs/[Helm] v3.0 or later is required.
* Deploy {longhorn-product-name} components images to your own registry.
* Deploy Kubernetes CSI driver components images to your own registry.

== Image Management for Air-Gapped Environments

. **Obtain the list of required images** +
Run the script to download the complete list of all images required for {longhorn-product-name}. The script will generate a `longhorn-images.txt` file containing the list of images.
+
[,bash]
----

> #!/usr/bin/env bash
> CHART_REF=oci://dp.apps.rancher.io/charts/suse-storage:{patch-version}
> set -euo pipefail
> setopt shwordsplit 2>/dev/null || true
> # Check for dependencies
> for cmd in yq jq helm tar; do
>   if ! command -v "$cmd" >/dev/null 2>&1; then
>     echo >&2 "Error: '$cmd' is required but not installed."
>     exit 1
>   fi
> done
> tmp_dir=$(mktemp -d)
> cleanup() {
>   rm -rf "$tmp_dir"
> }
> trap cleanup EXIT
> # Pull and extract chart
> helm pull "$CHART_REF" --untar --untardir "$tmp_dir"
> # The folder name is just the chart name, not the version
> chart_dir="$tmp_dir/suse-storage"
> chart_path="$chart_dir/Chart.yaml"
> # Extract 'helm.sh/images' annotation and convert to JSON
> images_json=$(
>   yq eval '.annotations."helm.sh/images"' "$chart_path" | yq -o=json eval -
> )
> # Directly print image list (one per line)
> echo "$images_json" | jq -r '.[] | .image' > longhorn-images.txt

Reviewer: ```suggestion
#!/usr/bin/env bash

CHART_REF=oci://dp.apps.rancher.io/charts/suse-storage:{patch-version}

set -euo pipefail
setopt shwordsplit 2>/dev/null || true

# Check for dependencies
for cmd in yq jq helm tar; do
  if ! command -v "$cmd" >/dev/null 2>&1; then
    echo >&2 "Error: '$cmd' is required but not installed."
    exit 1
  fi
done

tmp_dir=$(mktemp -d)
cleanup() {
  rm -rf "$tmp_dir"
}
trap cleanup EXIT

# Pull and extract chart
helm pull "$CHART_REF" --untar --untardir "$tmp_dir"

# The folder name is just the chart name, not the version
chart_dir="$tmp_dir/suse-storage"
chart_path="$chart_dir/Chart.yaml"

# Extract 'helm.sh/images' annotation and convert to JSON
images_json=$(
  yq eval '.annotations."helm.sh/images"' "$chart_path" | yq -o=json eval -
)

# Directly print image list (one per line)
echo "$images_json" | jq -r '.[] | .image' > longhorn-images.txt
```

Reviewer: Hello @sushant-suse 
I will be off today. Could you leave questions here? We can discuss offline.

    registrySecret: <REGISTRY_SECRET_NAME>

> ----

Reviewer 3: `defaultSettings.registrySecret: <SECRET_NAME>`  not exist in `values.yaml`([ref](https://github.com/longhorn/website/pull/1154/files)) Additionally, the example is missing the `privateRegistry.createSecret` setting.
